controllers/movies.js

```diff
@@ -0,0 +1,48 @@
+const movies = require('../movies')
+
+const getAllMovies = (request, response) => {
+  return response.send(movies)
+}
+
+const getByTitleOrDirector = (request, response) => {
+  const { input } = request.params
+
+  const foundTitleAndDirector = movies.filter((movie) => {
+    return movie.title.toLowerCase().includes(input) ||
+      movie.directors.toString().toLowerCase().includes(input)
+  })
+
+  return response.send(foundTitleAndDirector)
+}
+
+/* const directorMatch = (request, response) => {
+  const { directorOnly } = request.params
+
+  // eslint-disable-next-line max-len
+  const theDirector = movies.filter((movie) => {
+    return movie.directors.toString().toLowerCase().includes(directorOnly)
+  })
+
+  return response.send(theDirector)
+} */
+
+const saveNewMovie = (request, response) => {
+  const {
+    title, directors, releaseDate, rating, runTime, genres
+  } = request.body
+
+  if (!title || !directors || !releaseDate || !rating || !runTime || !genres) {
+    // eslint-disable-next-line max-len
+    return response.status(400).send('The following fields are required: location, mascot, abbreviation, conference, division')
+  }
+  const newMovie = {
+    // eslint-disable-next-line max-len
+    title: 'Cabin in the Woods', directors: '[Drew Goddard]', releaseDate: '04/13/2012', rating: 'R', runTime: '95 mins', genres: '[Horror]'
```

Comment:
_Nit_ these should not have values mapped to them. Its going to try and post the same movie every time instead of the one the user inputs. 
`const newMovie = { title, directors, releaseDate, rating, runTime, genres } `